### DISCUSSION TITLE
Handle prerender page activated event

### DISCUSSION THREAD

**Discussion by njfamirm** [alpinejs/alpine#3810] [1. Help] [answered]
Hi

I'm using `speculationrules` script for prerender my pages. some pages using alpine to show something using `x-show`, but when pre-rendered page visible and activated, the getter don't call again

this is my code:
```html
<button x-show="!$store.cart.isEmpty"> </button>
```
```js
export const cartStorage = {
  data: {
    itemList: {
      product: [],
    },
  },

  get isEmpty(): boolean {
    for (const category in this.data.itemList) {
      if (this.data.itemList[category].length > 0) {
        return false;
      }
    }
    return true;
  },
};

alpine.store('cart', cartStorage);
```

how can handle this in alpine js for some functions/getters

thanks

**Comment by njfamirm** [comment]
I'd find a solution that's from @alimd, he's using [requestAnimationFrame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) for call function, so when the page is active, this request callback called.


```js
requestAnimationFrame(() => {
  alpine.store(name).load();
});
```

so when the load function is called, another function has been run again.

but any other solution exists?

**Comment by ekwoka** [comment]
What is speculation rules?

This should definitely be running it.

Can you explain more what the situation is?

**Comment by njfamirm** [comment, marked as answer]
I'm use [request animation frame](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame) for reloading data.

**Reply by ekwoka** [reply]
Yes, I know what that is.

You shouldn't be using it for this.

But I don't see what any of your code has to do with each other?
